src/main/java/core/training/TrainingManager.java

```diff
@@ -179,79 +177,182 @@ public void recalcSubskills(boolean showBar) {
         }
     }
 
-    //----------------------------------- Utility Methods ----------------------------------------------------------
+	public void calculateTraining(Timestamp nextTrainingDate,
+								  List<TrainingPerWeek> trainingList,
+								  List<Player> currentPlayers,
+								  List<Player> previousPlayers,
+								  int trainerSkill,
+								  List<StaffMember> staff) {
+
+		// Generate a map of players from the previous hrf.
+		final Map<Integer, Player> players = new HashMap<>();
+		for (Player p : previousPlayers) {
+			players.put(p.getSpielerID(), p);
+		}
+
+		// Train each player
+		for (Player player : currentPlayers) {
+			try {
+
+				// The version of the player from last hrf
+				Player old = players.get(player.getSpielerID());
+				if (old == null) {
+					if (TrainingManager.TRAININGDEBUG) {
+						HOLogger.instance().debug(HOModel.class, "Old player for id " + player.getSpielerID() + " = null");
+					}
+					// Player appears the first time
+					// - was bought new
+					// - promoted from youth
+					// - it is the first hrf ever loaded
+					old = new Player();
+					old.setSpielerID(player.getSpielerID());
+					old.copySkills(player);
+					if (HOVerwaltung.instance().getModel().getCurrentPlayer(player.getSpielerID()) != null) {
+						// PLayer is in current team (not an historical player)
+						List<TrainingEvent> events = player.downloadTrainingEvents();
```

Comment:
is this the line which is trigered at staturp and cause match download ?  with the modifictation to match downloader would that stillwork correctly ?

Reply:
see above - but i'm not sure if this method will also call the OnlineWorker

Reply:
ok let just put a todo as mentioned above and we can leave it as is for the time being ....

Reply:
@akasolace line is removed